src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Parser.cs

```diff
@@ -0,0 +1,265 @@
+                        attributeSymbol.ContainingType.ToDisplayString() == RegexGeneratorAttributeName)
+                    {
+                        return methodDeclarationSyntax;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        // Returns null if nothing to do, Diagnostic if there's an error to report, or RegexType if the type was analyzed successfully.
+        private static object? GetRegexTypeToEmit(Compilation compilation, MethodDeclarationSyntax methodSyntax, CancellationToken cancellationToken)
+        {
+            // TODO: Use https://github.com/dotnet/runtime/pull/59092
+            INamedTypeSymbol? regexSymbol = compilation.GetTypeByMetadataName(RegexName);
+            INamedTypeSymbol? regexGeneratorAttributeSymbol = compilation.GetTypeByMetadataName(RegexGeneratorAttributeName);
+            if (regexSymbol is null || regexGeneratorAttributeSymbol is null)
+            {
+                // Required types aren't available
+                return null;
+            }
+
+            TypeDeclarationSyntax typeDec = methodSyntax.Parent as TypeDeclarationSyntax;
+            if (typeDec is null)
+            {
+                return null;
+            }
+
+            SemanticModel sm = compilation.GetSemanticModel(methodSyntax.SyntaxTree);
+
+            IMethodSymbol regexMethodSymbol = sm.GetDeclaredSymbol(methodSyntax, cancellationToken) as IMethodSymbol;
+            if (regexMethodSymbol is null)
+            {
+                return null;
+            }
+
+            ImmutableArray<AttributeData>? boundAttributes = regexMethodSymbol.GetAttributes();
+            if (boundAttributes is null || boundAttributes.Value.Length == 0)
+            {
+                return null;
+            }
+
+            DiagnosticDescriptor? errorDescriptor = null;
+            RegexMethod? regexMethod = null;
+            foreach (AttributeData attributeData in boundAttributes)
+            {
+                // If we already encountered an error, stop looking at this method's attributes.
+                if (errorDescriptor is not null)
+                {
+                    break;
+                }
+
+                // If this isn't a RegexGeneratorAttribute, skip it.
+                if (!attributeData.AttributeClass.Equals(regexGeneratorAttributeSymbol))
+                {
+                    continue;
+                }
+
+                if (attributeData.ConstructorArguments.Any(ca => ca.Kind == TypedConstantKind.Error))
+                {
+                    errorDescriptor = DiagnosticDescriptors.InvalidRegexGeneratorAttribute;
+                    break;
+                }
+
+                ImmutableArray<TypedConstant> items = attributeData.ConstructorArguments;
+                if (items.Length is > 0 and <= 3 && items[0].Value is string pattern)
```

Comment:
What if named arguments is used and the arguments order is no longer matching parameter order?

Reply:
It works fine.  I've added a test to confirm.